addons/pos_sale/static/src/app/order_management_screen/sale_order_management_screen/sale_order_management_screen.js

```diff
@@ -135,32 +131,31 @@ export class SaleOrderManagementScreen extends Component {
         } catch {
             // FIXME Universal catch seems ill advised
         }
-        const order_partner = this.pos.db.get_partner_by_id(sale_order.partner_id[0]);
+        const order_partner = this.pos.models["res.partner"].get(sale_order.partner_id);
         if (order_partner) {
             currentPOSOrder.set_partner(order_partner);
         } else {
             try {
-                await this.pos._loadPartners([sale_order.partner_id[0]]);
+                await this.pos._loadPartners([sale_order.partner_id]);
             } catch {
                 const title = _t("Customer loading error");
-                const body = _t(
-                    "There was a problem in loading the %s customer.",
-                    sale_order.partner_id[1]
-                );
+                const body = _t("There was a problem in loading the %s customer.");
                 this.dialog.add(AlertDialog, { title, body });
             }
-            currentPOSOrder.set_partner(this.pos.db.get_partner_by_id(sale_order.partner_id[0]));
+            currentPOSOrder.set_partner(this.pos.models["res.partner"].get(sale_order.partner_id));
         }
         const orderFiscalPos = sale_order.fiscal_position_id
-            ? this.pos.fiscal_positions.find(
+            ? this.pos.models["account.fiscal.position"].find(
                   (position) => position.id === sale_order.fiscal_position_id[0]
```

Comment:
```suggestion
                  (position) => position.id === sale_order.fiscal_position_id
```
Verify if it should be like this.

Reply:
This file is very inconsistent, sometimes you do `product_id` without accessing the first element, sometimes you do `pricelist_id[0]`. Which should it be?